### PR TITLE
DL-4929 Co-op social housing feature flag functionality

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-4929 Co-op social housing feature flag functionality

**New feature**

* Added feature flag to enable 2020 onwards change of 'Social housing' to 'Provider of social housing or housing co-operative'

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/models/summaryReturnsModels.scala [hunk 1]
```diff
@@ -90,6 +90,8 @@ case class SummaryReturnsModel(atedBalance: Option[BigDecimal] = None,
                               )
 
 object SummaryReturnsModel {
+  val newSocialHousingDescription = "Provider of social housing or housing co-operative"
```
Comment (paulanderson1234) [inline app/models/summaryReturnsModels.scala:93]: Should this be hardcoded text in the Model or utilise the message file?

File: app/models/summaryReturnsModels.scala [hunk 2]
```diff
@@ -98,11 +100,32 @@ object SummaryReturnsModel {
...
+              if (reliefReturn.reliefType == "Social housing") {
+                reliefReturn.copy(reliefType = newSocialHousingDescription)
+              } else {
+                reliefReturn
+              }
+            )
+          )), draftReturns = returnsForPeriod.draftReturns.map {
+            case draftReturn @ DraftReturns(periodKey, _, "Social housing", _, "Relief") if periodKey >= 2020 =>
+              draftReturn.copy(description = newSocialHousingDescription)
+            case draftReturn =>
```
Comment (Sir-Iyke) [inline app/models/summaryReturnsModels.scala:116]: Does this need an underscore case?

File: app/models/summaryReturnsModels.scala [hunk 3]
```diff
@@ -112,14 +135,34 @@ object SummaryReturnsModel {
...
+            if (reliefReturn.reliefType == newSocialHousingDescription) {
+              reliefReturn.copy(reliefType = "Social housing")
+            } else {
+              reliefReturn
+            }
+          )
+          )), draftReturns = returnsForPeriod.draftReturns.map {
+          case draftReturn @ DraftReturns(periodKey, _, `newSocialHousingDescription`, _, "Relief") if periodKey >= 2020 =>
+            draftReturn.copy(description = "Social housing")
+          case draftReturn =>
```
Comment (Sir-Iyke) [inline app/models/summaryReturnsModels.scala:154]: Does this need an underscore case?

File: app/models/summaryReturnsModels.scala (hunk 2 above)
Comment (jmountain-hmrc) [reply app/models/summaryReturnsModels.scala:116]: It is one, it's just bound to a variable

File: app/models/summaryReturnsModels.scala (hunk 1 above)
Comment (jmountain-hmrc) [reply app/models/summaryReturnsModels.scala:93]: It's being checked against an english text entry so for now I think I can be hardcoded (or moved to config)

File: test/utils/AtedUtilsSpec.scala [hunk 4]
```diff
@@ -23,6 +23,7 @@ import org.scalatestplus.mockito.MockitoSugar
 import org.scalatestplus.play.PlaySpec
 import org.scalatestplus.play.guice.GuiceOneServerPerSuite
 import play.api.test.FakeRequest
+import config.featureswitch.FeatureSwitch
```
Comment (Sir-Iyke) [inline test/utils/AtedUtilsSpec.scala:26]: I think this import can be deleted

**Pull request merged**